### ISSUE TITLE
Building AutoEncoder - Requesting for help and clarification

### ISSUE THREAD

**Issue by zollen** [deeplearning4j/deeplearning4j#7899]
I had a very simple AutoEncoder program. I created some dummy INDArray data. I was expecting the inputs are same as the actual output (at least very close). But unfortunately they are not. I think I might have misunderstood how deeplearning4j works... I would be much appreciated if anyone could tell me what I did wrong. Thanks!!!!!
 
```
public static void main(String [] args) throws Exception {
		
		INDArray expected = Nd4j.zeros(new int[] { 5, 4 });
		
		for (int i = 0; i < 5; i++) {
			for (int j = 0; j < 4; j++) {
				expected.putScalar(new int[] { i, j }, i + j);
			}
		}
			
        System.out.println("Building model....");
		MultiLayerConfiguration conf = new NeuralNetConfiguration.Builder()
				.seed(83)
				.cacheMode(CacheMode.HOST)
				.weightInit(WeightInit.XAVIER)
				.updater(new Nadam())
				.activation(Activation.TANH)
				.optimizationAlgo(OptimizationAlgorithm.STOCHASTIC_GRADIENT_DESCENT)
				.l2(0.0001)
				.list()
				.layer(0, new AutoEncoder.Builder().nIn(4).nOut(2).build())
				.layer(1, new OutputLayer.Builder().nIn(2).nOut(4)
								.lossFunction(LossFunctions.LossFunction.MSE).build())
				
				.build();
		
		MultiLayerNetwork network = new MultiLayerNetwork(conf);
		network.init();
				
		network.setListeners(new ScoreIterationListener(1));
		
		System.out.println(network.summary());
		
		for (int i = 0; i < 30; i++) {
			network.fit(expected, expected);
		}

		INDArray actual = network.output(expected, false);
		
		System.out.println(expected);
		System.out.println("========================");
		System.out.println(actual);
}
```

* Deeplearning4j version - 1.0.0-beta4
* platform information - Windows 10, JDK 11

**Comment by AlexDBlack** [comment]
There's a number of issues here.
1. You haven't set an activation function on the output layer. The default is likely not appropriate.
2. 30 iterations isn't nearly enough on most datasets
3. You need to set - and tune - the learning rate. Don't use the default learning rate, it's usually not appropriate.
4. The range of your input values is bad (too large). Your data should be normalized.
5. A bottleneck of 2 values (nOut=2) isn't appropriate
6. You're using an autoencoder layer, but calling fit - which is for supervised training. If you want to pretrain the autoencoder layer, use pretrain/pretrainLayer methods instead. Also read the javadoc for both the fit and pretrain methods.

Have a look at these pages:
https://deeplearning4j.org/docs/latest/deeplearning4j-nn-visualization
https://deeplearning4j.org/docs/latest/deeplearning4j-troubleshooting-training
